client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/Sections/Order.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { ReactElement, useMemo, useState } from 'react';
+import {
+  Autocomplete,
+  Box,
+  InputWithLabelRow,
+  NumericTextInput,
+  NumUtils,
+  Typography,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { DraftRequestLine } from '../hooks';
+
+interface OrderProps {
+  disabled?: boolean;
+  isPacksEnabled?: boolean;
+  draft?: DraftRequestLine | null;
+  update: (patch: Partial<DraftRequestLine>) => void;
+}
+
+export const Order = ({
+  disabled,
+  isPacksEnabled,
+  draft,
+  update,
+}: OrderProps): ReactElement => {
+  const t = useTranslation();
+  const [itemType, setItemType] = useState('units');
+
+  const options = useMemo(() => {
+    if (!isPacksEnabled) {
+      return [{ label: t('label.units'), value: 'units' }];
+    }
+    return [
+      { label: t('label.units'), value: 'units' },
+      { label: t('label.packs'), value: 'packs' },
+    ];
+  }, [isPacksEnabled, t]);
+
+  const defaultOption = options.find(option => option.value === itemType);
+
+  const calculatePackQuantity = (precision: number = 0): number => {
+    return NumUtils.round(
+      (draft?.requestedQuantity ?? 0) / (draft?.defaultPackSize ?? 1),
+      precision
+    );
+  };
+
+  const alternativeQuantityLabel = useMemo((): string => {
+    if (itemType === 'packs') {
+      return t('label.order-count-units', {
+        count: Math.ceil(draft?.requestedQuantity ?? 0),
+      });
+    }
+    return t('label.order-count-packs', { count: calculatePackQuantity() });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [itemType, draft?.requestedQuantity]);
+
+  const currentValue = useMemo(() => {
+    if (itemType === 'packs') return calculatePackQuantity(2);
+    return Math.ceil(draft?.requestedQuantity ?? 0);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [itemType, draft?.requestedQuantity, draft?.defaultPackSize]);
+
+  const handleValueChange = (value?: number) => {
+    const newValue = isNaN(Number(value)) ? 0 : (value ?? 0);
+    const requestedQuantity =
+      itemType === 'packs'
+        ? newValue * (draft?.defaultPackSize ?? 0)
+        : newValue;
```

Comment:
Would love to see these helper methods extracted into own file with unit tests :eyes: